packages/sources/elwood/src/adapter.ts

```diff
@@ -0,0 +1,110 @@
+import { AxiosRequestConfig, Builder, Logger, Validator } from '@chainlink/ea-bootstrap'
+import {
+  AdapterRequest,
+  APIEndpoint,
+  Config,
+  ExecuteWithConfig,
+  ExecuteFactory,
+  MakeWSHandler,
+  Requester,
+} from '@chainlink/ea-bootstrap'
+import { makeConfig } from './config'
+import * as endpoints from './endpoint'
+import { PriceResponse, ResponseMessage, SubscribeRequest } from './types'
+
+export const execute: ExecuteWithConfig<Config, endpoints.TInputParameters> = async (
+  request,
+  context,
+  config,
+) => {
+  return Builder.buildSelector<Config, endpoints.TInputParameters>(
+    request,
+    context,
+    config,
+    endpoints,
+  )
+}
+
+export const endpointSelector = (
+  request: AdapterRequest,
+): APIEndpoint<Config, endpoints.TInputParameters> =>
+  Builder.selectEndpoint<Config, endpoints.TInputParameters>(request, makeConfig(), endpoints)
+
+export const makeExecute: ExecuteFactory<Config, endpoints.TInputParameters> = (config) => {
+  return async (request, context) => execute(request, context, config || makeConfig())
+}
+
+const getPair = (input: AdapterRequest) => {
+  const validator = new Validator(input, endpoints.price.inputParameters, {})
+  const base = validator.validated.data.base.toUpperCase()
+  const quote = validator.validated.data.quote.toUpperCase()
+  return `${base}-${quote}`
+}
+
+const getSubscribeRequest = (
+  pairSymbol: string,
+  action: 'subscribe' | 'unsubscribe',
+): SubscribeRequest => ({
+  action: action,
+  stream: 'index',
+  symbol: pairSymbol,
+  index_freq: 1_000, //ms
+})
+
+export const makeWSHandler =
+  (
+    config?: Config,
+  ): MakeWSHandler<
+    ResponseMessage | any // TODO: full WS message types
```

Comment:
Can we get this TODO resolved before merging?

Reply:
If the response types that are used in the adapter code are all included here, I don't think we need to go and get every possible ws message for the API, as long as the code itself is typesafe I'm good

Reply:
I can add `util.isObject` checks to ensure we at least know the input is an object before trying to operate on it